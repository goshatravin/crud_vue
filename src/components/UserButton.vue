<template>
    <div class="home">
        <button v-on:click="UserPopUpForm"></button>
    </div>
</template>

<script>
export default {

};
</script>

<style>

</style>
