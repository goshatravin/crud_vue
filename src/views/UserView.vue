<template>
    <div class="userView">
        <UserDashboard/>
        <UserButton/>
        <UserPopUpForm/>
    </div>
</template>

<script>
import UserButton from '@/components/UserButton.vue';
import UserDashboard from '@/components/UserDashboard.vue';
import UserPopUpForm from '@/components/UserPopUpForm.vue';

export default {
    name: 'UserView',
    components: {
        UserButton,
        UserDashboard,
        UserPopUpForm,
    },
};
</script>

<style>

</style>
