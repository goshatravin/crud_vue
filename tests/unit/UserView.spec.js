import { shallowMount } from '@vue/test-utils';
import UserView from '@/views/UserView.vue';
import UserButton from '@/components/UserButton.vue';
import UserDashboard from '@/components/UserDashboard.vue';
import UserPopUpForm from '@/components/UserPopUpForm.vue';

describe('UserView', () => {
    const mockFn = jest.fn();
    const build = () => {
        const wrapper = shallowMount(UserView);
        const userBtn = shallowMount(UserButton);
        const formq = shallowMount(UserPopUpForm);
        return {
            formq,
            wrapper,
            userBtn,
            creatButton: () => wrapper.find(UserButton),
            userPlace: () => wrapper.find(UserDashboard),
            // findButtom: () => userBtn.find('button'),
            PopUpForm: () => wrapper.find(UserPopUpForm),
        };
    };
    it('renders the component', () => {
        // arrange
        const { wrapper } = build();
        // assert
        expect(wrapper.html()).toMatchSnapshot();
    });
    it('renders main child components', () => {
        // arrange
        const { creatButton, userPlace, findButtom } = build();
        // assert
        expect(creatButton().exists()).toBe(true);
        expect(userPlace().exists()).toBe(true);
        expect(findButtom().exists()).toBe(true);
    });
    it('click on the btn success', () => {
        const { userBtn } = build();
        userBtn.find('button').trigger('click');
        expect(mockFn).toHaveBeenCalled();
    });
});
