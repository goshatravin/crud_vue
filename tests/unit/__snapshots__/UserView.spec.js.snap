// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders the component 1`] = `
<div class="userView">
  <userdashboard-stub></userdashboard-stub>
  <userbutton-stub></userbutton-stub>
  <userpopupform-stub></userpopupform-stub>
</div>
`;
